src/group/managed_group/mod.rs

```diff
@@ -635,14 +646,22 @@ impl<'a> ManagedGroup<'a> {
         self.active
     }
 
-    /// Sets a different `CredentialBundle`
-    pub fn set_credential_bundle(&mut self, credential_bundle: &'a CredentialBundle) {
-        self.credential_bundle = credential_bundle;
+    /// Returns own credential. If the group is inactive, it returns a
+    /// `UseAfterEviction` error.
+    pub fn credential(&self) -> Result<Credential, ManagedGroupError> {
+        if !self.is_active() {
+            return Err(ManagedGroupError::UseAfterEviction(UseAfterEviction::Error));
+        }
+        let tree = self.group.tree();
+        Ok(tree.own_key_package().credential().clone())
     }
 
-    /// Returns own credential
-    pub fn credential(&self) -> &Credential {
-        &self.credential_bundle.credential()
+    /// Tries to obtain the credential bundle corresponding to the group's
+    /// credential from the key store.
+    pub fn credential_bundle(&self) -> Result<&CredentialBundle, ManagedGroupError> {
```

Comment:
Do we really need this for something?

Reply:
It was meant as a convenience function, but I'm removing it with a follow-up PR anyway, since it doesn't play nicely with the `RefCell` approach. So I can remove it already an fetch it "manually" from the key store every time, or I leave it in until the next PR. Alternatively, if this PR doesn't land today, I'm going to close it anyway, so I can continue developing through next week. We can then land everything in one big go the week after next.